### PR TITLE
[InstallAPI] Add support for aliased exports

### PR DESCRIPTION
**Pull request by cyndyishida**
Apple's ld supports alias_lists, described as
```
     -alias_list filename
             The specified filename contains a list of aliases. The symbol name and its alias are on one
             line, separated by whitespace.  Lines starting with # are ignored.
```
To handle this for installapi-produced TBD files, pass along the same input and account for it in verification.

### CODE REVIEW

File: clang/lib/InstallAPI/DylibVerifier.cpp [hunk 1]
```diff
@@ -973,5 +978,24 @@ bool DylibVerifier::verifyBinaryAttrs(const ArrayRef<Target> ProvidedTargets,
   return true;
 }
 
+std::unique_ptr<SymbolSet> DylibVerifier::getExports() {
+  for (const auto &[Alias, Base] : Aliases) {
```
Comment (ributzka) [inline clang/lib/InstallAPI/DylibVerifier.cpp:982]: This loop runs every time someone calls `getExports()`. Could this be cached, so it only happens once?

File: clang/tools/clang-installapi/Options.h [hunk 2]
```diff
@@ -107,6 +107,9 @@ struct LinkerOptions {
   /// \brief Additional library search paths.
   PathSeq LibPaths;
 
+  /// \brief List of aliased symbol exports.
```
Comment (ributzka) [inline clang/tools/clang-installapi/Options.h:110]:
```suggestion
  /// \brief List of alias symbol files.
```

File: clang/lib/InstallAPI/DylibVerifier.cpp (hunk 1 above)
Comment (cyndyishida) [reply clang/lib/InstallAPI/DylibVerifier.cpp:982]: `getExports()` only gets called once for the whole lifetime of InstallAPI as it releases ownership over the exports to forward along to write out the final TBD file.

Comment (ributzka) [reply clang/lib/InstallAPI/DylibVerifier.cpp:982]: In that case I would update the function name to `takeExports`.

**Comment by cyndyishida** [comment]
ping, is there anything else needed to land this patch?

File: clang/tools/clang-installapi/Options.cpp [hunk 3]
```diff
@@ -259,7 +259,10 @@ bool Options::processLinkerOptions(InputArgList &Args) {
   if (auto *Arg = Args.getLastArg(drv::OPT_umbrella))
     LinkerOpts.ParentUmbrella = Arg->getValue();
 
-  LinkerOpts.IsDylib = Args.hasArg(drv::OPT_dynamiclib);
```
Comment (zixu-w) [inline clang/tools/clang-installapi/Options.cpp:262]: Is this expected to get removed?

Comment (cyndyishida) [reply clang/tools/clang-installapi/Options.cpp:262]: nope, looks like I was missing some test coverage, added one implicitly here https://github.com/llvm/llvm-project/pull/88750/files#diff-0b1bbd3584706760de8a4030db1b04120c2c7c1f640681be80114f59278988bfR22

**Review by zixu-w — APPROVED** [review]

**Pull request merged**